libfwupdplugin/fu-bytes.c

```diff
@@ -281,3 +281,28 @@ fu_bytes_get_data_safe(GBytes *bytes, gsize *bufsz, GError **error)
 	}
 	return buf;
 }
+
+/**
+ * fu_bytes_to_string:
+ * @bytes: data blob
+ *
+ * Converts @bytes to a lowercase hex string.
+ *
+ * Returns: (transfer full): a string, which may be zero length
+ *
+ * Since: 2.0.4
+ **/
+gchar *
+fu_bytes_to_string(GBytes *bytes)
+{
+	const guint8 *buf;
+	gsize bufsz = 0;
+	g_autoptr(GString) str = g_string_new(NULL);
```

Comment:
Wouldn't it better to avoid `g_autoptr` and `g_steal_pointer` with the snipplet below?
```
GString *str = NULL;
g_return_val_if_fail(bytes != NULL, NULL);
str = g_string_new(NULL);
```

Reply:
Sure, it's just one extra line. :) The reason we normally use the autoptr for trivial cases like this is that trivial functions often become more complicated (or where we derestrict things like non-null), and then we leak the GString if we return early for an error.